doc/tutorials/error_analysis/error_analysis_part1.ipynb

```diff
@@ -0,0 +1,459 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Tutorial: Error Estimation - Part 1 (Introduction and Binning Analysis)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Table of contents\n",
+    "0. [Data generation](#Data-generation)\n",
+    "1. [Introduction](#Introduction)\n",
+    "2. [Uncorrelated samples](#Uncorrelated-samples)\n",
+    "3. [Binning analysis](#Binning-analysis)\n",
+    "4. [References](#References)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Data generation\n",
+    "\n",
+    "In this tutorial, you will learn how to estimate the accuracy of your simulation results. Because we are going to emply statistical methods, we need a fair amount of data to play with. The following code cell will generate two data sets which will be used throughout the tutorial."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# generate simulation data using AR(1) processes\n",
+    "\n",
+    "print(\"Generating data sets for the tutorial ...\\n\")\n",
+    "\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "np.random.seed(43)\n",
+    "\n",
+    "def ar_1_process(n_samples, y0, c, phi, eps, n_warmup):\n",
+    "    y = y0\n",
+    "    for i in range(n_warmup):\n",
+    "        y = c + phi*y + eps*np.random.uniform(-1.0,1.0)\n",
+    "    ys = np.zeros(n_samples)\n",
+    "    ys[0] = y\n",
+    "    for i in range(1,n_samples):\n",
+    "        ys[i] = c + phi*ys[i-1] + eps*np.random.uniform(-1.0,1.0)\n",
+    "    return ys\n",
```

Comment:
You guessed right about the warmup phase. The reason why it is included is because the mean value of the series is unknown, or at least I haven't come across a simple formula how to compute it from the input parameters. So without the warmup phase, there would be a bias in the first part of the data range.
I agree on this method being slow, but it is no concern here. Executing the cell is almost instantaneously. Would it not be good to have something rather verbose in the tutorial, so the trainee can understand what's going on? I, for one, find the idea of an AR(1) process quite simple and intuitive, but would be deterred to find out more about the process if a more abstract "scipy.signal.lfilter" method would be called. And if there are bugs in the code, would the following error analysis not detect them and report them in the tests?
I agree that joining the two loops would be a good idea in order to not have duplicate code. Other than that, I think it is a matter of taste?

Reply:
> The reason why it is included is because the mean value of the series is unknown, or at least I haven't come across a simple formula how to compute it from the input parameters. So without the warmup phase, there would be a bias in the first part of the data series.

Good point, the bias is actually visible in the time series with a warmup of 0.

> And if there are bugs in the code, would the following error analysis not detect them and report them in the tests?

Hopefully. We once had a flipped sign in a cross product in the C++ core ([diff](https://github.com/espressomd/espresso/pull/2382/commits/5945d070b7037d98d88fb2fe3efa3055c77ef1b7#diff-d45782ab798b1f002476e4ac52098301d1bf8d74ab0de0b17c04229fef1570a6L335-L343)). The test was always green. It took 4 years for someone to notice.

I've reverted my changes to your original implementation (00dbceb5ad5c1a055813fe9f1c5da497cbcc27c3) and moved the scipy function to the test case (4cf00de72f16503c231ab7d12a4e79e72586422f).

Reply:
Isn't the mean value of the AR-1 process mu=c/(1-phi) for |phi|<1 ?https://en.m.wikipedia.org/wiki/Autoregressive_model

Reply:
> Isn't the mean value of the AR-1 process mu=c/(1-phi) for |phi|<1 ?https://en.m.wikipedia.org/wiki/Autoregressive_model

Good hint. Using the expressions there, it is possible to omit the warmup phase and the initial value y_0. 

> moved the scipy function to the test case ([4cf00de](https://github.com/espressomd/espresso/commit/4cf00de72f16503c231ab7d12a4e79e72586422f)).

Thanks. Hmmm... well it is annoying that the two notebooks are supposed to have some identical code, so they require identical tests.
One obvious solution would be to join the two tutorials together. Or just live with the duplicate code.

Reply:
We could move the function to a python file and import it in the notebooks, but then users would have to manually open the python file if they want to see the function implementation...